### PR TITLE
Refactor: Rename deploy_to_cv role and module and release as cv_deploy

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Rename deploy_to_cv role and module.

`arista.avd.deploy_to_cv` *module* -> `arista.avd.cv_workflow`
`arista.avd.deploy_to_cv` *role* -> `arista.avd.cv_deploy`

## How to test

Verified manually that old playbooks got deprecation warnings for role and module.
And for module:

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/cv_deploy/README.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+Depending on the configured options, the role supports multiple operations:
+
+- Deploys configurations for one or more devices using the "Static Configuration Studio".
+- Deploys device and interface Tags for one or more devices.
+- Adds missing devices and updates device details for existing devices in the "Inventory & Topology Studio".
+- Creates, builds, submits Workspaces.
+- Creates, approves, starts Change Controls.
+- Deploys special metadata for CV Pathfinder solution.
+
+Devices will be identified using `serial_number`, `system_mac_address` or `hostname` (in prioritized order).
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/cv_deploy/README.md:26]:
```suggestion
Devices will be identified using `serial_number`, `system_mac_address` or `hostname` (in precedence order).
```

File: ansible_collections/arista/avd/roles/cv_deploy/README.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+cv_skip_missing_devices: true
+```
+
+#### Role behavior configuration
+
+By default the role will
+
+1. Create a workspace.
+2. Push all configurations and tags.
+3. Unassign tags
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/cv_deploy/README.md:169]: is this in the correct order ?

**Review by noredistribution — COMMENTED** [review]
where running the action module I do get

```
[DEPRECATION WARNING]: arista.avd.deploy_to_cv has been deprecated. The preview module 'arista.avd.deploy_to_cv' has been
released under the new name 'arista.avd.cv_workflow'. The module will be redirected automatically until AVD version 5.0.0
after which it will be removed. This feature will be removed from arista.avd in version 5.0.0. Deprecation warnings can be
disabled by setting deprecation_warnings=False in ansible.cfg.
```

when using the role I get it during the task

```
TASK [arista.avd.deploy_to_cv : Deprecation warning!] ************************************************************************
Wednesday 27 March 2024  14:07:10 +0000 (0:00:05.648)       0:00:26.069 *******
Wednesday 27 March 2024  14:07:10 +0000 (0:00:05.648)       0:00:26.069 *******
ok: [tp-avd-spine1 -> localhost] =>
  msg:
  - '!!! Deprecation Warning - The preview role ''arista.avd.deploy_to_cv'' has been released under the new name ''arista.avd.cv_deploy''.'
  - '!!! Deprecation Warning - The role will be redirected automatically until AVD version 5.0.0 after which it will be removed.'
  - '!!! Deprecation Warning - Importing role: arista.avd.cv_deploy ...'
  - 
 ```
  
but I also get a playbook failure when printing the old `deploy_to_cv_results`

```
fatal: [tp-avd-leaf6]: FAILED! =>
  msg: |-
    The task includes an option with an undefined variable. The error was: 'deploy_to_cv_results' is undefined. 'deploy_to_cv_results' is undefined
 ```

That's expected I assume?
This is my playbook task just in case
```    - name: Deploy configurations and tags to CloudVision
      tags: [deploy]
      ansible.builtin.import_role:
        name: arista.avd.deploy_to_cv
      vars:
        cv_server: www.cv-prod-uk-1.arista.io
        cv_token: "{{ lookup('file', 'token.tok')}}" 
        deploy_to_cv_accept_preview: true
        cv_skip_missing_devices: true
        cv_submit_workspace: true
        cv_run_change_control: true
        cv_configlet_name_template: "AVD-${hostname}"
        cv_register_detailed_results: true
        cv_change_control_description: "Tamas test Ansible AVD new deploy to cv role"
        cv_verify_certs: false
    - name: debug
      debug:
        msg: "{{deploy_to_cv_results}}"```

**Review by gmuloc — APPROVED** [review]
LGTM from a code point of view
Module redirection works well
I have not tested the role redirection nor the interaction with CV

**Comment by ClausHolbechArista** [comment]
@noredistribution 
`but I also get a playbook failure when printing the old deploy_to_cv_results`
Good catch! I renamed the registered variable, but I should handle that in the redirection. I will look into it.

**Comment by gmuloc** [comment]
> where running the action module I do get
> 
> ```
> [DEPRECATION WARNING]: arista.avd.deploy_to_cv has been deprecated. The preview module 'arista.avd.deploy_to_cv' has been
> released under the new name 'arista.avd.cv_workflow'. The module will be redirected automatically until AVD version 5.0.0
> after which it will be removed. This feature will be removed from arista.avd in version 5.0.0. Deprecation warnings can be
> disabled by setting deprecation_warnings=False in ansible.cfg.
> ```
> 
> when using the role I get it during the task
> 
> ```
> TASK [arista.avd.deploy_to_cv : Deprecation warning!] ************************************************************************
> Wednesday 27 March 2024  14:07:10 +0000 (0:00:05.648)       0:00:26.069 *******
> Wednesday 27 March 2024  14:07:10 +0000 (0:00:05.648)       0:00:26.069 *******
> ok: [tp-avd-spine1 -> localhost] =>
>   msg:
>   - '!!! Deprecation Warning - The preview role ''arista.avd.deploy_to_cv'' has been released under the new name ''arista.avd.cv_deploy''.'
>   - '!!! Deprecation Warning - The role will be redirected automatically until AVD version 5.0.0 after which it will be removed.'
>   - '!!! Deprecation Warning - Importing role: arista.avd.cv_deploy ...'
>   - 
> ```
> 
> but I also get a playbook failure when printing the old `deploy_to_cv_results`
> 
> ```
> fatal: [tp-avd-leaf6]: FAILED! =>
>   msg: |-
>     The task includes an option with an undefined variable. The error was: 'deploy_to_cv_results' is undefined. 'deploy_to_cv_results' is undefined
> ```
> 
> That's expected I assume? This is my playbook task just in case
> 
> ```
>       tags: [deploy]
>       ansible.builtin.import_role:
>         name: arista.avd.deploy_to_cv
>       vars:
>         cv_server: www.cv-prod-uk-1.arista.io
>         cv_token: "{{ lookup('file', 'token.tok')}}" 
>         deploy_to_cv_accept_preview: true
>         cv_skip_missing_devices: true
>         cv_submit_workspace: true
>         cv_run_change_control: true
>         cv_configlet_name_template: "AVD-${hostname}"
>         cv_register_detailed_results: true
>         cv_change_control_description: "Tamas test Ansible AVD new deploy to cv role"
>         cv_verify_certs: false
>     - name: debug
>       debug:
>         msg: "{{deploy_to_cv_results}}"```
> ```

This has changed indeed:

https://github.com/aristanetworks/avd/pull/3785/files#diff-8357ca6bfbfe632dd9f227192ffd549b40a0886a2e6a91e5a8a63ea47ed2cb82R56

**Review by noredistribution — APPROVED** [review]
tested both the old names and new names against 2024.1.0 and CVaaS and didn't break any of my playbooks

**Pull request merged**